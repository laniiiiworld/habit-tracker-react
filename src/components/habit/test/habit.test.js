import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Habit from '../habit';

describe('Habit Component', () => {
  const habit = { id: 1, name: 'Testing', count: 4 };
  let HabitComponent;
  let onIncrement;
  let onDecrement;
  let onDelete;

  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();
    HabitComponent = (
      <Habit //
        habit={habit}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
      />
    );
  });

  it('renders', () => {
    const component = renderer.create(HabitComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('Button click', () => {
    beforeEach(() => {
      render(HabitComponent);
    });

    it('calls onIncrement when clicking the "increment" button', () => {
      const button = screen.getByTitle('increment');
      userEvent.click(button);
      expect(onIncrement).toHaveBeenCalledWith(habit);
    });

    it('calls onDecrement when clicking the "decrement" button', () => {
      const button = screen.getByTitle('decrement');
      userEvent.click(button);
      expect(onDecrement).toHaveBeenCalledWith(habit);
    });

    it('calls onDelete when clicking the "delete" button', () => {
      const button = screen.getByTitle('delete');
      userEvent.click(button);
      expect(onDelete).toHaveBeenCalledWith(habit);
    });
  });
});
