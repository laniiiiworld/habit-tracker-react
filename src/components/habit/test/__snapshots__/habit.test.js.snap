// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habit Component renders 1`] = `
<li
  className="habit"
>
  <span
    className="habit-name align-left"
    data-testid="habit-name"
  >
    Testing
  </span>
  <span
    className="align-right"
  >
    <span
      className="count habit-count"
      data-testid="habit-count"
    >
      4
    </span>
    <button
      className="habit-button habit-increase"
      onClick={[Function]}
      title="increment"
    >
      <i
        className="fa-solid fa-square-plus"
      />
    </button>
    <button
      className="habit-button habit-decrease"
      onClick={[Function]}
      title="decrement"
    >
      <i
        className="fa-solid fa-square-minus"
      />
    </button>
    <button
      className="habit-button habit-delete"
      onClick={[Function]}
      title="delete"
    >
      <i
        className="fa-solid fa-trash-can"
      />
    </button>
  </span>
</li>
`;
